sys/vane/clay.hoon

```diff
@@ -2314,6 +2360,66 @@
         ?^(r.mun ~ !!) :: [~ %w !>([t.yak (forge-nori yak)])])-all
       (query(ank.dom ank:(descend-path:(zu ank.dom) r.mun)) p.mun) ::  dead code
     ::
+    ::  Gets the permissions that apply to a particular node.
+    ::
+    ::  If the node has no permissions of its own, we use its parent's.
+    ::  If no permissions have been set for the entire tree above the node,
+    ::  we default to fully private (empty whitelist).
+    ::
+    ++  read-p
+      |=  pax/path
+      ^-  (unit (unit (each cage lobe)))
+      =-  [~ ~ %& %noun !>(-)]
+      :-  (read-p-in pax per.red)
+      (read-p-in pax pew.red)
+    ::
+    ++  read-p-in
+      |=  {pax/path pes/(map path rule)}
+      ^-  dict
+      =+  rul=(~(get by pes) pax)
+      ?^  rul  [pax u.rul]
+      ?~  pax  [/ %white ~]
+      $(pax t.pax)
+    ::
+    ++  may-read
+      |=  {who/ship car/care yon/aeon pax/path}
+      ^-  ?
+      ?+  car
+        (allowed-by who pax per.red)
+      ::
+          $p
+        =(who our)
+      ::
+          ?($y $z)
+        =+  tak=(~(get by hit.dom) yon)
+        ?~  tak  |
+        =+  yak=(tako-to-yaki u.tak)
+        =+  len=(lent pax)
+        =-  (levy ~(tap in -) |=(p/path (allowed-by who p per.red)))
+        %+  roll  ~(tap in (~(del in ~(key by q.yak)) pax))
```

Comment:
This seems likely to be a duplicate of the usual %y / %z logic. If so it might be good to make a little helper function.

Reply:
It's not an exact match. `%y` wants just the path, `%z` wants lobes too, and they both want only the path tails, where the permission check just wants the full paths.  
The logic it still fairly transferable though, the data structures here are easy enough to work with. Having one function for "get child nodes with tail paths" seems to give a net gain to simplicity.

Reply:
Actually, I'm having trouble finding a helper function here that isn't in some way terrible to use for two out of three cases. Problem is that if we do a dir listing for `/one` with the following data stored, then we still want to see `./two` show up, but we can't provide a `lobe` alongside that, unlike for `./three`

```
/one/two/x
/one/two/y
/one/three
```

Reply:
Ok, no need to force it if it doesn't feel right.